<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <script>
      const promiseOne = new Promise(function (resolve, reject) {
        setTimeout(function () {
          // console.log("task one is in process");
          resolve();
        }, 1000);
      });

      promiseOne.then(function () {
        // console.log("promise one is completed");
      });

      new Promise(function (resolve, reject) {
        setTimeout(function () {
          // console.log("task 2 is in process");
          resolve();
        }, 2000);
      }).then(function () {
        // console.log("task is consumed");
      });

      const promiseFour = new Promise(function (resolve, reject) {
        setTimeout(function () {
          // console.log("promise four is in process");

          let error = true;

          if (!error) {
            resolve({ username: "Sufyan Sial", email: "Sufyansufi@123.com" });
          } else {
            reject("Error: Something went wrong");
          }
        }, 1000);
      });

      promiseFour
        .then(function (user) {
          // console.log(user);
          return user.username;
        })
        .then(function (username) {
          // console.log(username);
        })
        .catch(function (err) {
          // console.log(err);
        });
      // .finally(() =>
      //   console.log("The promise is either resolved or rejected")
      // );

      const promiseFive = new Promise(function (resolve, reject) {
        setTimeout(function () {
          let error = false;
          if (!error) {
            resolve({ username: "sufyan bin munir", email: "123@123.COM" });
          } else {
            reject("Error: something is off");
          }
        }, 1000);
      });

      async function consumingPromiseFive() {
        try {
          const data = await promiseFive;
          console.log(data);
        } catch (error) {
          console.log(error);
        }
      }

      consumingPromiseFive();

      // async function getAllUsers() {
      //   try {
      //     const response = await fetch(
      //       "https://api.github.com/users/hiteshchoudhary"
      //     );
      //     const dataa = await response.json();
      //     console.log(dataa);
      //   } catch (error) {
      //     console.log(error);
      //   }
      // }
      // getAllUsers();

      fetch("https://api.github.com/users/hiteshchoudhary")
        .then((response) => {
          response.json();
        })
        .then((data) => {
          console.log(data);
        })
        .catch((error) => {
          console.log(error);
        });
    </script>
  </body>
</html>
